src/AnalyzeTrait.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\analyze;
+
+use Drupal\Core\Config\ImmutableConfig;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+
+/**
+ * Provides shared service and functions to use across analyze entities.
+ */
+trait AnalyzeTrait {
```

Comment:
I wonder if this would more appropriate as a service rather than a trait? Makes dependencies much more straight forward etc.

Reply:
Agreed: will implement.